libraries/botbuilder-dialogs/src/skillDialog.ts

```diff
@@ -0,0 +1,140 @@
+        // Send of of conversation to the skill if the dialog has been cancelled. 
+        if (reason == DialogReason.cancelCalled || reason == DialogReason.replaceCalled) {
+            await context.sendTraceActivity(`${ this.id }.EndDialogAsync()`, undefined, undefined, `ActivityType: ${ context.activity.type }`);
+
+            const reference = TurnContext.getConversationReference(context.activity);
+            // Apply conversation reference and common properties from incoming activity before sending.
+            const activity = TurnContext.applyConversationReference({ type: ActivityTypes.EndOfConversation }, reference, true);
+            activity.channelData = context.activity.channelData;
+
+            await this.sendToSkill(context, activity as Activity);
+        }
+
+        await super.endDialog(context, instance, reason);
+    }
+
+    /**
+     * Clones the Activity entity.
+     * @param activity Activity to clone.
+     */
+    private cloneActivity(activity: Partial<Activity>): Activity {
+        return Object.assign({} as Activity, activity);
+    }
+
+    private static validateBeginDialogArgs(options: any): SkillDialogArgs {
+        if (!options) {
+            throw new TypeError('Missing options parameter');
+        }
+
+        const dialogArgs = options as SkillDialogArgs;
+
+        if (!dialogArgs.activity) {
+            throw new TypeError(`"activity" is undefined or null in options.`);
+        }
+
+        // Only accept Message or Event activities
+        if (dialogArgs.activity.type !== ActivityTypes.Message && dialogArgs.activity.type !== ActivityTypes.Event) {
+            // Just forward to the remote skill
+            throw new TypeError(`Only ${ ActivityTypes.Message } and ${ ActivityTypes.Event } activities are supported. Received activity of type ${ dialogArgs.activity.type } in options.`);
+        }
+
+        return dialogArgs;
+    }
+
+    private async sendToSkill(context: TurnContext, activity: Activity): Promise<void> {
+        // Create a conversationId to interact with the skill and send the activity
+        const skillConversationId = await this.dialogOptions.conversationIdFactory.createSkillConversationId(TurnContext.getConversationReference(activity) as ConversationReference);
+
+        // Always save state before forwarding
+        // (the dialog stack won't get updated with the skillDialog and things won't work if you don't)
+        await this.dialogOptions.conversationState.saveChanges(context, true);
+        const skillInfo = this.dialogOptions.skill;
+        const response = await this.dialogOptions.skillClient.postActivity(this.dialogOptions.botId, skillInfo.appId, skillInfo.skillEndpoint, skillInfo.skillEndpoint, skillConversationId, activity);
+
+        // Inspect the skill response status
+        if (!(response.status >= 200 && response.status <= 299)) {
```

Comment:
dont we have constants for this? Or otherwise it might be worth checking if in other places we do something like this which owuld be 'isSuccessStatusCode'. C# has a built in method for this actually, given how frequently folks ask this question. Potentially something to think about in general rather than for this PR, definitely would not block on it.